server/api-util/lineItemHelpers.js

```diff
@@ -268,11 +268,14 @@ exports.constructValidLineItems = lineItems => {
  * @returns boolean
  */
 exports.hasCommissionPercentage = commission => {
-  const percentage = commission.percentage;
+  const percentage = commission?.percentage;
   const isDefined = percentage != null;
   const isNumber = typeof percentage === 'number' && !isNaN(percentage);
   if (isDefined && !isNumber) {
     throw new Error(`${percentage} is not a number.`);
   }
-  return isDefined;
+
+  // Only create a line item if the percentage is not zero
+  const isZero = percentage === 0;
+  return isDefined && !isZero;
```

Comment:
It's a number at this point, so perhaps we should use 'greater than'. At least for me, it's more understandable that the commission should be more than zero (instead of not-zero).

```suggestion
  // Only create a line item if the percentage is set to be more than zero
  const isMoreThanZero = percentage > 0;
  return isDefined && isMoreThanZero;
```

Reply:
Fair point, updated!